cmd/hubagent/workload/setup.go

```diff
@@ -188,7 +188,10 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
 		defaultSchedulingQueue := queue.NewSimpleClusterResourcePlacementSchedulingQueue()
 		defaultScheduler := scheduler.NewScheduler("DefaultScheduler", defaultFramework, defaultSchedulingQueue, mgr)
 		klog.Info("Starting the scheduler")
-		defaultScheduler.Run(ctx)
+		// Scheduler must run in a separate goroutine as Run() is a blocking call.
+		go func() {
```

Comment:
when will this go routine dangle on exit? Do we not want to monitor/wait for it?

Reply:
Hi Ryan! The scheduler is set to exit on context cancellation (same for the controller manager).

Reply:
I will add a waitgroup to make things clearer for both.